### PR TITLE
New yank behaviour blog post

### PR DESCRIPTION
**Pull request by arthurnn**
Blog post for https://github.com/rubygems/rubygems.org/pull/923

review @sferik @indirect @qrush @dwradcliffe 

This is the first draft. Blog post is small, I think we can keep it like that. 
Maybe add a FAQ ? not sure if it is necessary tho.

### CODE REVIEW

File: _posts/2015-04-13-permadelete-on-yank.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: Policy change: Yank will now delete the .gem file
+layout: post
+author: Arthur Nogueira Neves
+author_email: arthurnn@gmail.com
+---
+
+The gem yank action, is a command gem authors can call to yank a version out of the rubygems index. There was some expectation that the yank command would also delete the .gem file, however that was not the case, and people would just open a request on help.rubygems.org to actually delete the gem. For a while we only deleted gems manually, using a script, but that was getting out of hand, as the script had to be ran manually.
```
Comment (sferik) [inline _posts/2015-04-13-permadelete-on-yank.md]: I wouldn’t use the word “yank” to define the behavior of `gem yank`. Consider “remove” instead.

Comment (sferik) [inline _posts/2015-04-13-permadelete-on-yank.md]: You say “manually” twice in the last sentence.

**Comment by qrush** [comment]
So this is pretty rough. I took the structure @arthurnn laid out here, rewrote and elaborated a bit. Sadly refactoring doesn't work as well with prose. How does this sound? (Also, is it possible to get the Markdown out of a comment after being posted?)

# Heads up: Policy change about `gem yank`

`gem yank` now deletes the gem permanently as of __DATE__. (Finally!)

## Why?

"Yanking" a gem was introduced as way to remove a gem from the RubyGems.org index, usually because of an accidental `gem push` or someone wanting to allow others to take over the gem's name.

Our policy has been to not permanently delete the .gem file when a gem was yanked, but over the years that has caused massive support backlogs. Most of the support cases are due to someone accidentally pushing a gem that has internal or sensitive code - and usually our team of volunteers can take weeks to months to respond and remove it.

One note about the new behavior: `gem yank` will remove the gem from S3 and our CDN, but it will not stop unofficial mirrors from copying the gem or anyone publicly downloading [via webhooks](http://guides.rubygems.org/rubygems-org-api/#webhook-methods). If you've pushed a gem with internal code, you still need to reset API keys, URLs, or anything else sensitive despite the new behavior.

## A final note

A concern of ours about `gem yank` and its behavior was that someone could maliciously or accidentally remove gems that others depended on. However, we've been using an Amazon S3 bucket to store the gems for years now [with versioning](http://docs.aws.amazon.com/AmazonS3/latest/dev/Versioning.html) on - so if someone does remove gems that are necessary, we can easily restore them if necessary. We hope it won't ever be.

Hopefully we'll see this reduce [our support load](http://help.rubygems.org). Thanks for bearing with us if you've had to deal with `gem yank`'s behavior in the past.

**Comment by indirect** [comment]
If you're a collab, you can edit the post and then copy the markdown directly out of it.

```
# Changes to the yank command

*tl;dr* Today, when you yank a gem, we remove the gem from the main list of gems, but we don't delete the .gem file. Starting on __DATE__, yanking a gem will also delete the .gem file and it will not be possible to download that version of the gem anymore.

## Yanking without deletion

A very long time ago, @qrush created the `gem yank` command so that gem authors could remove a gem from the RubyGems.org main index of all gems. Some authors pushed gems accidentally, some authors pushed gems that had critical bugs, and some authors just wanted to remove their gem so someone else could have the name.

When the `yank` command was created, yanking would prevent a gem from being installed with `gem install`, but it did not remove the .gem file hosted on RubyGems.org. Anyone sufficiently interested or motivated could manually download the .gem file, and even manually install it after that.

Over the years, this has created a huge backlog of support requests for the unpaid volunteers running RubyGems.org. At this point, the vast majority of the support requests for RubyGems.org are gem authors who have published passwords or private code, yanked the gem, and then discovered that the .gem file is still possible to download.

Because deleting gems completely was a slow, manual process, it could take weeks or even months before the volunteer support team could handle a gem deletion request. In order to reduce this support backlog, we're going to change how the yank command works. Starting on __DATE__, all yanked gems will not just be removed from the RubyGems index, they will also be deleted from the RubyGems.org server. 

Keep in mind, however, that even though the yanked gem will be deleted from the RubyGems.org servers, the gem could easily have been downloaded by anyone who runs their own unofficial mirror of RubyGems.org. If you push a gem that has sensitive URLs, API keys, passwords, or anything else, you still need to change them right away.

One reason that we previously kept the .gem files even when gems were yanked was so that even hacking a gem author's account wouldn't be enough to remove gems completely. We've recently changed how we store gems, and we will be able to manually restore gems that have been deleted via yank if that ever becomes necessary.

We hope that this change will reduce [our support load](http://help.rubygems.org), and make it easier for gem authors to manage their own gems themselves without having to get help from the RubyGems.org volunteers.
```

File: _posts/2015-04-13-permadelete-on-yank.md (hunk 1 above)
Comment (bf4) [inline _posts/2015-04-13-permadelete-on-yank.md]: also, for consistently, `s/authors/owners`

**Comment by bf4** [comment]
So, does yanking all versions of a gem make that gem name available to others?

**Comment by bf4** [comment]
If you yank `2.10.0`, can you push another `2.10.0`?

**Comment by bf4** [comment]
Wasn't there a scenario where if you have a yanked gem in your Gemfile.lock, bundler will install it from S3?

**Comment by indirect** [comment]
@bf4 we were planning to do that, but if the gems are deleted, we won't implement it. :)

**Comment by dwradcliffe** [comment]
> If you yank 2.10.0, can you push another 2.10.0?

Nope. Same policy as always applies to re-pushing a specific version.

**Comment by qrush** [comment]
We should clarify that policy has *not* changed.

**Comment by arthurnn** [comment]
@qrush I copied/pasted your version, added a few missing words and added the last statment. let me know your final thoughts.

thanks (would like to merge this before railsconf)

**Comment by dwradcliffe** [comment]
:+1:
